tools/mason/MasonPublish.chpl

```diff
@@ -99,64 +93,136 @@ proc masonPublish(args: [] string) throws {
+  try! {
+    if registryPath == MASON_HOME then return true;
+    if !exists('.git') {
+      throw new owned MasonError(registryPath + ' is not a local git repository.');
+    }
+    if trueIfLocal {
+      if exists(registryPath) && exists(registryPath + "/Bricks/") {
+        return true;
+      }
+      else {
+        throw new owned MasonError(registryPath + " is not a valid path to a local mason-registry.");
+        exit(0);
+      }
+    }
+    else {
+      var command = ('git ls-remote ' + registryPath).split();
+      var checkRemote = spawn(command, stdout=PIPE);
+      checkRemote.wait();
+      if checkRemote.exit_status == 0 then return true;
+      else {
+        throw new owned MasonError(registryPath + " is not a valid remote path");
+        exit(0);
+      }
+    }
+  }
+  catch e {
+    writeln(e.message());
+    exit(0);
+  }
+}
 /* Main Script that goes through the act of publishing the package to the mason registry.
    Takes the package owners GitHub username as input will throw errors through command
    line git commands if any of the git calls fails.
  */
-proc publishPackage(username: string) throws {
-  const packageLocation = absPath(here.cwd());
-  var stream = makeRandomStream(int);
-  var uniqueDir = stream.getNext(): string;
-  const name = getPackageName();
-  const safeDir = MASON_HOME + '/tmp/' + name + '-' + uniqueDir;
+proc publishPackage(username: string, registryPath : string, isLocal : bool) throws {
   try! {
-    if !exists(MASON_HOME + '/tmp') then mkdir(MASON_HOME + '/tmp');
-    mkdir(safeDir);
-    cloneMasonReg(username, safeDir);
-    branchMasonReg(username, name, safeDir);
-    addPackageToBricks(packageLocation, safeDir, name);
-    gitC(safeDir + "/mason-registry", "git add .");
-    gitC(safeDir + "/mason-registry", "git commit -m '" + name + "'");
-    gitC(safeDir + "/mason-registry", 'git push --set-upstream origin ' + name, true);
-    rmTree(safeDir + '/');
-    writeln('--------------------------------------------------------------------');
-    writeln('Go to the above link to open up a Pull Request to the mason-registry');
+    const packageLocation = absPath(here.cwd());
+    var stream = makeRandomStream(int);
+    var uniqueDir = stream.getNext(): string;
+    const name = getPackageName();
+    var safeDir = '';
+
+    if isLocal then safeDir = registryPath;
+    else {
+      safeDir = MASON_HOME + '/tmp/' + name + '-' + uniqueDir;
+    }
+
+    if !isLocal {
+      if !exists(MASON_HOME + '/tmp') then mkdir(MASON_HOME + '/tmp');
+      mkdir(safeDir);
+    }
+
+    if !isLocal {
+      cloneMasonReg(username, safeDir, registryPath);
+      branchMasonReg(username, name, safeDir, registryPath);
+    }
+
+    addPackageToBricks(packageLocation, safeDir, name, registryPath, isLocal);
+
+    if !isLocal {
+      gitC(safeDir + "/mason-registry", "git add .");
+      gitC(safeDir + "/mason-registry", "git commit -m '" + name + "'");
```

Comment:
I think we want a little more info in the commit message and maybe something that tells us the commit was auto-generated, such as `Adding <name>@<version> package to registry via mason publish`